### PR TITLE
[`from_pretrained`]  Simpler code for peft

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Simplifies the logic when loading a PEFT model: the `_adapter_model_path` is used instead of `maybe_has_adapter_model_path` and `has_adapter_config`

### CODE REVIEW

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -3221,9 +3211,9 @@ def from_pretrained(
         if quantization_method_from_config == QuantizationMethod.GPTQ:
             model = quantizer.post_init_model(model)
 
-        if has_adapter_config:
+        if _adapter_model_path:
```
Comment (sgugger) [inline src/transformers/modeling_utils.py:3214]:
```suggestion
        if _adapter_model_path is not None:
```

**Review by sgugger — APPROVED** [review]
Looks simpler indeed!

**Review by younesbelkada — COMMENTED** [review]
Thanks a lot @ArthurZucker for attempting to refactor that part, sadly all PEFT tests are failing with your changes, can you please make sure they pass before merging?

```bash
RUN_SLOW=1 pytest tests/peft_integration/test_peft_integration.py
```

Also make sure to rebase with main in order to run the tests

I suspect the reason is because you need to override the `pretrained_model_name_or_path` instance with `_adapter_model_path` and store the current adapter id in a new variable `adapter_model_id` in the previous changes

**Comment by ArthurZucker** [comment]
Thanks! Can confirm that the tests are now green! @younesbelkada some are really fast do you not want to remove the `slow` mention for them?

**Review by younesbelkada — APPROVED** [review]
Thanks! All of them should be using tiny models so they are fast, I don't know if we want to add PEFT as a dependency to our CIs so maybe let's keep it as it is for now, what do you think?

**Pull request merged**